### PR TITLE
Feat: add `oauthGrantType` option to support `response_type` code

### PR DESCRIPTION
**Pull request by ashabottu**
added oauthGrantType - to determine whether the response_type is code or implicit

### CODE REVIEW

File: packages/ra-auth-cognito/src/authProvider.ts [hunk 1]
```diff
@@ -221,11 +222,15 @@ export const CognitoAuthProvider = (
                             'profile',
                             'aws.cognito.signin.user.admin',
                         ];
+                        const response_type =
+                            oauthOptions.oauthGrantType === 'code'
+                                ? 'code'
+                                : 'token';
```
Comment (djhi) [inline packages/ra-auth-cognito/src/authProvider.ts:228]: What about `implicit`?

Comment (djhi) [reply packages/ra-auth-cognito/src/authProvider.ts:228]: oh I see now. Thanks

**Review by djhi — APPROVED** [review]

**Review by alexpop89 — APPROVED** [review]

**Comment by alexpop89** [comment]
@djhi can we please merge this in a new version?

**Comment by JakeStoeffler** [comment]
+1, I need this as well. Willing to help. What would it take to get this across the finish line @djhi ?

**Review by slax57 — APPROVED** [review]

**Comment by slax57** [comment]
@ashabottu @alexpop89 @JakeStoeffler I just released this feature in the latest major version (2.0.0).
This version also includes the upgrade to react-admin v5.
Thanks again for your contribution!

**Pull request merged**